### PR TITLE
DOC: Refactor _create_delegator_method using functools

### LINKED ISSUE #59783: DOC: Questions on Usage of *args, **kwargs, and inplace parameters in pandas.Series.cat methods
### Pandas version checks

- [X] I have checked that the issue still exists on the latest versions of the docs on `main` [here](https://pandas.pydata.org/docs/dev/)

### Location of the documentation

https://pandas.pydata.org/docs/reference/api/pandas.Series.cat.remove_categories.html#pandas.Series.cat.remove_categories

https://pandas.pydata.org/docs/reference/api/pandas.Series.cat.rename_categories.html#pandas.Series.cat.rename_categories

### Documentation problem

I am currently trying to work on the issue #59592, specifically for the remove_categories and rename_categories methods. Before finalizing my changes, I need some clarification on a few points regarding the documentation:

1. `*args` and `**kwargs` Usage: 
- For remove_categories, the method only seems to use the removals parameter. 
- Similarly, for rename_categories, `*args` and `**kwargs` are mentioned but not utilized in the method implementation. Could you clarify whether the inclusion of `*args` and `**kwargs` in the documentation for methods is by design? 
- I have noticed this pattern across many methods and wanted to confirm if this is intentional or if the documentation should be updated to reflect actual usage.

 2. `inplace` Parameter: The `inplace` parameter for methods is included in the error codes of the validation docstring. However, it is not explicitly mentioned in the current documentation, nor is it included as part of the method definition for both methods.

### Suggested fix for documentation

Verify usage of ` **args`, `**kwargs`, and 'inplace' if not make changes to the `pandas.Series.cat` methods in docs.

### PR DESCRIPTION
**Pull request by saldanhad**
* Refactoring `_create_delegator_method` using `functools`, partially fixes the docstrings for below under #59592 and fixes #59783 :
```
        -i "pandas.Series.cat.add_categories PR01,PR02" \
        -i "pandas.Series.cat.as_ordered PR01" \
        -i "pandas.Series.cat.as_unordered PR01" \
        -i "pandas.Series.cat.remove_categories PR01,PR02" \
        -i "pandas.Series.cat.remove_unused_categories PR01" \
        -i "pandas.Series.cat.rename_categories PR01,PR02" \
        -i "pandas.Series.cat.reorder_categories PR01,PR02" \
        -i "pandas.Series.cat.set_categories PR01,PR02" \
```
```
        -i "pandas.Series.dt.to_period PR01,PR02" \
        -i "pandas.Series.dt.as_unit PR01,PR02" \
        -i "pandas.Series.dt.ceil PR01,PR02" \
        -i "pandas.Series.dt.day_name PR01,PR02" \
        -i "pandas.Series.dt.floor PR01,PR02" \
        -i "pandas.Series.dt.month_name PR01,PR02" \
        -i "pandas.Series.dt.normalize PR01" \
        -i "pandas.Series.dt.round PR01,PR02" \
        -i "pandas.Series.dt.strftime PR01,PR02" \
        -i "pandas.Series.dt.total_seconds PR01" \
        -i "pandas.Series.dt.tz_convert PR01,PR02" \
        -i "pandas.Series.dt.tz_localize PR01,PR02" \
```

### CODE REVIEW

File: pandas/core/accessor.py [hunk 1]
```diff
@@ -116,13 +116,15 @@ def _setter(self, new_values):
                 doc=getattr(delegate, accessor_mapping(name)).__doc__,
             )
 
+        import functools as ft
```
Comment (yuanx749) [inline pandas/core/accessor.py:119]: Very elegant solution for the signatures. Just curious, why not put the import at the top of the file？

Comment (saldanhad) [reply pandas/core/accessor.py:119]: Felt it would help with readability this way, have seen similar imports elsewhere as well.

File: pandas/core/arrays/datetimes.py [hunk 2]
```diff
@@ -261,6 +261,7 @@ def _scalar_type(self) -> type[Timestamp]:
     )
     _datetimelike_methods: list[str] = [
         "to_period",
+        "to_timestamp",
```
Comment (rhshadrach) [inline pandas/core/arrays/datetimes.py:264]: This method should not be added to `DatetimeArray`; it already exists on `PeriodArray`.

However you do want a different docstring for `Series.dt.to_datetime` than the one on `PeriodArray`. I think this is a systemic issue that impacts all `dt` methods. We can use `ft.wraps` as you are doing here, but then overwrite the docstring. We will need a place to put the docstrings, and then write one for each method in the `dt` namespace. This seems to me to be a large effort, and should not be undertaken before being thoroughly discussed.

If there are better, low effort, solutions than the above, I'm certainly willing to entertain them here. But if not, I think we can move forward with this PR after removing the changes to this file.

**Review by rhshadrach — CHANGES_REQUESTED** [review]
Thanks for the PR!

File: pandas/core/arrays/datetimes.py (hunk 2 above)
Comment (saldanhad) [reply pandas/core/arrays/datetimes.py:264]: Thank you for the review. I agree that the `dt` methods need separate handling as you have suggested. Will revert these changes along with the other suggested edits for this PR.

File: pandas/core/arrays/categorical.py [hunk 3]
```diff
@@ -1371,8 +1377,8 @@ def remove_categories(self, removals) -> Self:
         """
         Remove the specified categories.
 
-        `removals` must be included in the old categories. Values which were in
-        the removed categories will be set to NaN
+        ``removals`` must be included in the old categories,
```
[the remark this answers is not in the document]
Comment (saldanhad) [reply pandas/core/arrays/categorical.py:1380]: I could rephrase it to like: The  ``removals`` argument must be a subset of the current categories.

File: pandas/core/accessor.py (hunk 1 above)
Comment (saldanhad) [reply pandas/core/accessor.py:119]: sure

**Review by rhshadrach — APPROVED** [review]
lgtm

File: pandas/core/accessor.py [hunk 4]
```diff
@@ -7,6 +7,7 @@
 
 from __future__ import annotations
 
+import functools as ft
```
[the remark this answers is not in the document]
Comment (saldanhad) [reply pandas/core/accessor.py:10]: If I don't use ft then I will have to format the decorator as @functools.wrap. However, using ft is the standard approach and helps keep it concise.

[the remark this answers is not in the document]
Comment (mroeschke) [reply pandas/core/accessor.py:10]: > If I don't use ft then I will have to format the decorator as @functools.wrap

Yup that's fine. We don't use the `as ft` convention anywhere in pandas (nor do I see this convention in the Python docs)

[the remark this answers is not in the document]
Comment (saldanhad) [reply pandas/core/accessor.py:10]: oh I see, will implement the changes.

**Comment by saldanhad** [comment]
Just inquiring to see if i've missed to implement any changes here ? @mroeschke

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @saldanhad
